packages/components/src/test/tree/controlled/UseHierarchyLevelFiltering.test.ts

```diff
@@ -208,4 +209,41 @@ describe("useHierarchyLevelFiltering", () => {
     result.current.clearFilter(parentNode.item);
     expect(modelSource.getModel().getNode(childNode.id)).to.be.undefined;
   });
+
+  it("clears filter unsubscribes from observable created by `applyFilter`", () => {
+    const applyFilterActionSubject = new Subject<TreeNodeLoadResult>();
+    const clearFilterActionSubject = new Subject<TreeNodeLoadResult>();
+    let isFilterCalled = false;
+    nodeLoaderMock.setup((x) => x.loadNode(moq.It.isAny(), 0)).returns(() => isFilterCalled ? clearFilterActionSubject : applyFilterActionSubject);
```

Comment:
I think it should be possible to remove `isFilterCalled` variable:

```suggestion
    nodeLoaderMock.setup((x) => x.loadNode(moq.It.isAny(), 0)).returns(() => applyFilterActionSubject);
    nodeLoaderMock.setup((x) => x.loadNode(moq.It.isAny(), 0)).returns(() => clearFilterActionSubject);
```